src/main/scala/where/Condition.scala

```diff
@@ -36,13 +36,21 @@ case class NullCondition(reference: Reference, check: NullCheck) extends Conditi
     reference.getReferenceable.toSet
 }
 
-case class PredicateCondition(predicate: Predicate, projection: Collection, where: AnyNode => Condition) extends Condition {
+case class PredicateCondition(
+  predicate: Predicate,
+  projection: Collection,
+  where: ObjectReference => Where
+) extends Condition {
   def toQuery(referenceableMap: ReferenceableMap): String = {
     val identifier = "x"
-    val node = AnyNode()
-    val fakeMap: ReferenceableMap = Map(node -> identifier)
-    val conditionString = where(node).toQuery(fakeMap)
-    s"${predicate.toQuery} ($identifier in ${projection.toQuery(referenceableMap)} where $conditionString)"
+    val referenceable = AnyNode()
+    val adjustedMap = referenceableMap + (referenceable -> identifier)
```

Comment:
also, since i'm adding this to `referenceableMap` now (you can reference other things in your conditions), there's no guarantee that `identifier` won't collide with a key that's already in there. it won't in practice, since identifiers are created like `a#`, i duno, i don't think it's worth making this more robust right now. 

1 possible solution is to make `ReferenceableMap` it's own class that handles generating fresh identifiers. then line 47 becomes `val adjustedMap = referenceableMap add referenceable`. this wouldn't be so hard, and since i'm already using the type alias it means i would avoid having to do a bunch of updates.